mm2src/coins_activation/src/eth_with_token_activation.rs

```diff
@@ -0,0 +1,234 @@
+            .iter()
+            .fold(BigDecimal::from(0), |total, (_, addr_info)| {
+                &total + &addr_info.balances.get_total()
+            })
+    }
+}
+
+impl CurrentBlock for EthWithTokensActivationResult {
+    fn current_block(&self) -> u64 { self.current_block }
+}
+
+#[async_trait]
+impl PlatformWithTokensActivationOps for EthCoin {
+    type ActivationRequest = EthWithTokensActivationRequest;
+    type PlatformProtocolInfo = CoinProtocol;
+    type ActivationResult = EthWithTokensActivationResult;
+    type ActivationError = EthActivationV2Error;
+
+    async fn enable_platform_coin(
+        ctx: MmArc,
+        ticker: String,
+        platform_conf: Json,
+        activation_request: Self::ActivationRequest,
+        protocol: Self::PlatformProtocolInfo,
+        priv_key: &[u8],
+    ) -> Result<Self, MmError<Self::ActivationError>> {
+        let platform_coin = eth_coin_from_conf_and_request_v2(
+            &ctx,
+            &ticker,
+            &platform_conf,
+            activation_request.platform_request,
+            priv_key,
+            protocol,
+        )
+        .await?;
+
+        Ok(platform_coin)
+    }
+
+    fn token_initializers(
+        &self,
+    ) -> Vec<Box<dyn TokenAsMmCoinInitializer<PlatformCoin = Self, ActivationRequest = Self::ActivationRequest>>> {
+        vec![Box::new(Erc20Initializer {
+            platform_coin: self.clone(),
+        })]
+    }
+
+    async fn get_activation_result(&self) -> Result<EthWithTokensActivationResult, MmError<EthActivationV2Error>> {
+        let my_address = self.my_address().map_err(EthActivationV2Error::InternalError)?;
+
+        let current_block = self
+            .current_block()
+            .compat()
+            .await
+            .map_err(EthActivationV2Error::InternalError)?;
+
+        let eth_balance = self
+            .my_balance()
+            .compat()
+            .await
+            .map_err(|e| EthActivationV2Error::CouldNotFetchBalance(e.to_string()))?;
+        let token_balances = self
+            .get_tokens_balance_list()
+            .await
+            .map_err(|e| EthActivationV2Error::CouldNotFetchBalance(e.to_string()))?;
+
+        let mut result = EthWithTokensActivationResult {
+            current_block,
+            eth_addresses_infos: HashMap::new(),
+            erc20_addresses_infos: HashMap::new(),
+        };
+
+        result
+            .eth_addresses_infos
+            .insert(my_address.to_string(), CoinAddressInfo {
+                derivation_method: DerivationMethod::Iguana,
+                pubkey: my_address.clone(),
+                balances: eth_balance,
+            });
+
+        result
+            .erc20_addresses_infos
+            .insert(my_address.to_string(), CoinAddressInfo {
+                derivation_method: DerivationMethod::Iguana,
+                pubkey: my_address,
+                balances: token_balances,
+            });
+        drop_mutability!(result);
```

Comment:
Is it really required to drop mutability just before returning?

Reply:
In most cases it's important to drop mutability so if the developer will change the value by mistake the project will fail at compilation. If you don't do such things, you might miss it on the review session and if there is no tests written for that unit that would cause critical bug.

But here, it's not necessary. So I will remove it.

Reply:
done